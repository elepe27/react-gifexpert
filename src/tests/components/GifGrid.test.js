import { shallow } from "enzyme/build";
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('pruebas en componente <GifGrid/>', () => {

    let categories = 'One Piece';
    
    test('debe cargar correctamente el componente', () => { 
        

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={ categories } />)
        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => { 

        
        const gifs = [{
            id:'ABC',
            url: 'https://localhost/foto.jpg'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={ categories } />);
        

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );


       
        
    });

    
    
});