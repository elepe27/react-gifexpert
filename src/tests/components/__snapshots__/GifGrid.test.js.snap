// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en componente <GifGrid/> debe cargar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en componente <GifGrid/> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      url="https://localhost/foto.jpg"
    />
  </div>
</Fragment>
`;
